src/main/java/org/opensearch/security/resources/api/migrate/MigrateResourceSharingInfoApiAction.java

```diff
@@ -245,22 +243,23 @@ private ValidationResult<Triple<String, Map<String, String>, List<SourceDoc>>> l
         clear.addScrollId(scrollId);
         client.clearScroll(clear).actionGet();
 
-        return ValidationResult.success(Triple.of(sourceIndex, typeToDefaultAccessLevel, results));
+        return ValidationResult.success(new ValidationResultArg(sourceIndex, defaultOwner, typeToDefaultAccessLevel, results));
     }
 
     /**
      * This method:
      *      1. Parses existing sharing info to a new ResourceSharing records
      *      2. Indexes the new record into corresponding resource-sharing index
      */
-    private ValidationResult<MigrationStats> createNewSharingRecords(Triple<String, Map<String, String>, List<SourceDoc>> sourceInfo)
-        throws IOException {
+    private ValidationResult<MigrationStats> createNewSharingRecords(ValidationResultArg sourceInfo) throws IOException {
         AtomicInteger migratedCount = new AtomicInteger();
-        AtomicReference<Set<String>> skippedNoUser = new AtomicReference<>();
-        skippedNoUser.set(new HashSet<>());
+        AtomicReference<Set<String>> skippedNoType = new AtomicReference<>();
```

Comment:
nit: consider encapsulating this within MigrationStats and call a fn with a name like `addNoType(String resourceId)`. We can also encapsulate the message that's created.

Reply:
should be fine as the scope if limited to this class. not worried about encapsulation as no other class will be able to reference MigrationStats directly.